src/YawPitchControl/YawPitchControl.js

```diff
@@ -634,33 +621,79 @@ export default class YawPitchControl extends Component {
 		}, duration);
 	}
 
-	get() {
-		return this.axes.get();
+	enableSensor() {
+		return new Promise((resolve, reject) => {
+			const activateSensor = () => {
+				if (this.options.gyroMode !== GYRO_MODE.NONE) {
+					this._deviceSensor.enable();
+					resolve();
+				} else {
+					reject(new Error("gyroMode not set"));
+				}
+			};
+
+			if (DeviceMotionEvent && typeof DeviceMotionEvent.requestPermission === "function") {
+				DeviceMotionEvent.requestPermission().then(permissionState => {
+					if (permissionState === "granted") {
+						activateSensor();
+					} else {
+						reject(new Error("Permission not granted"));
+					}
+				}).catch(e => {
+					// This can happen when this method was't triggered by user interaction
```

Comment:
@WoodNeck 

The following can be used like `.catch(reject)`

Reply:
I know, just want to add some nice looking comment on it so take this way.